src/plugins/intel_cpu/src/nodes/kernels/x64/random_uniform.cpp

```diff
@@ -626,10 +625,193 @@ void RandomUniform<isa>::tail(const std::vector<Vmm>& vmm_dst) {
     L(l_end);
 }
 
-template class RandomUniform<x64::avx512_core>;
-template class RandomUniform<x64::avx2>;
-template class RandomUniform<x64::sse41>;
+template class PhiloxGenerator<x64::avx512_core>;
+template class PhiloxGenerator<x64::avx2>;
+template class PhiloxGenerator<x64::sse41>;
+
+#undef GET_OFF
+
+//////////////// MERSENNE TWISTER GENERATOR ////////////////////
+
+#define GET_OFF(field) offsetof(MersenneTwisterGeneratorCallArgs, field)
+template <x64::cpu_isa_t isa>
+MersenneTwisterGenerator<isa>::MersenneTwisterGenerator(const MersenneTwisterGeneratorCompileParams& jcp) :
+    JitKernel(jit_name(), jcp, isa) {}
+
+template <x64::cpu_isa_t isa>
+void MersenneTwisterGenerator<isa>::generate() {
+    this->preamble();
+    registersPool = RegistersPool::create(isa, {rax, rcx, rsp, rdi, k0});
+
+    r64_dst = getReg64();
+    r64_state = getReg64();
+    r64_state_shift = getReg64();
+    r64_step = getReg64();
+    r64_work_amount = getReg64();
+    r64_elements_remaining = getReg64();
+
+    mov(r64_dst,  ptr[r64_params + GET_OFF(dst_ptr)]);
+    mov(r64_state_id, ptr[r64_params + GET_OFF(state_id)]);
+    mov(r64_state_shift, ptr[r64_params + GET_OFF(state_shift)]);
+    mov(r64_step, ptr[r64_params + GET_OFF(step)]);
+    mov(r64_work_amount, ptr[r64_params + GET_OFF(work_amount)]);
+    mov(r64_elements_remaining, ptr[r64_params + GET_OFF(elements_remaining)]);
+
+    initVectors();
+    process();
+
+    registersPool.reset();
+    this->postamble();
+}
+
+template <x64::cpu_isa_t isa>
+void MersenneTwisterGenerator<isa>::initVectors() {
+    const auto r64_aux = getReg64();
+    const auto r32_aux = Xbyak::Reg32(r64_aux.getIdx());
+
+    v_min = getVmm();
+    v_range = getVmm();
+    v_state = getVmm();
+    v_result = getVmm();
+    v_mask = getVmm();
+    v_divisor = getVmm();
+
+    // Initialize constants based on the requested data type
+    if (m_jcp.out_data_type == element::f32) {
+        BROADCAST_R(vpbroadcastd, v_mask, r32_aux, (1 << std::numeric_limits<float>::digits) - 1)
```

Comment:
That works for AVX512 only. Please run tests with [Intel SDE](https://www.intel.com/content/www/us/en/developer/articles/tool/software-development-emulator.html). Try at least -nhm, -skl, -spr

Reply:
Done!